docs/proposals/batch-node-process.md

```diff
@@ -0,0 +1,213 @@
+  ConfigFile string
+}
+
+// Config defines the batch-join config file format
+type Config struct {
+  Keadm     Keadm  `yaml:"keadm"`
+  Nodes     []Node `yaml:"nodes"`
+  MaxRunNum int    `yaml:"maxRunNum"`
+}
+
+type Keadm struct {
+  Download          Download          `yaml:"download"`
+  KeadmVersion      string            `yaml:"keadmVersion"`
+  ArchGroup         []string          `yaml:"archGroup"`
+  OfflinePackageDir *string           `yaml:"offlinePackageDir,omitempty"`
+  CmdTplArgs        map[string]string `yaml:"cmdTplArgs,omitempty"`
+}
+
+type Download struct {
+  URL    *string `yaml:"url,omitempty"`
+  Enable bool    `yaml:"enable"`
+}
+
+// Node defines the node information used in batch-join config file
+type Node struct {
+  NodeName string  `yaml:"nodeName"`
+  KeadmCmd string  `yaml:"keadmCmd"`
+  CopyFrom *string `yaml:"copyFrom,omitempty"`
+  SSH      SSH     `yaml:"ssh"`
+}
+
+type SSH struct {
+  IP       string     `yaml:"ip"`
+  Username string     `yaml:"username"`
+  Port     *int       `yaml:"port,omitempty"`
+  Auth     AuthConfig `yaml:"auth"`
+}
+
+type AuthConfig struct {
+  Type           string          `yaml:"type"`
+  PasswordAuth   *PasswordAuth   `yaml:"passwordAuth,omitempty"`
+  PrivateKeyAuth *PrivateKeyAuth `yaml:"privateKeyAuth,omitempty"`
+}
+
+type PasswordAuth struct {
+  Password string `yaml:"password"`
+}
+
+type PrivateKeyAuth struct {
+  PrivateKeyPath string `yaml:"privateKeyPath"`
+}
+```
+
+`BatchProcessOptions.ConfigFile`: Used to specify the configuration file for batch node processing. `example: keadm batch-process -c./config.yaml`
+
+`Config.Keadm`: Used to configure the download of the keadm installation package.
+
+- `Config.Keadm.Download`: When `Config.Keadm.Download.Enable` is `true`, the installation package will be downloaded from the address provided by `Config.Keadm.Download.URL`. If `URL` is empty, the installation package will be downloaded from the official github repository by default. When `Config.Keadm.Download.Enable` is `false`, the offline installation package provided by `OfflinePackageDir` will be used.
+- `Config.Keadm.KeadmVersion`: Specifies the version of the keadm installation package.
+- `Config.Keadm.ArchGroup`: Specifies the keadm architecture, and multiple architectures can be specified at the same time. Currently, the supported architectures are `arm`, `arm64`, and `amd64`.
+- `Config.Keadm.OfflinePackageDir`: Specifies the directory of the offline keadm installation package.
+- `Config.Keadm.CmdTplArgs`: The general template for the keadm execution command, used in conjunction with `Config.Nodes[x].KeadmCmd`.
+
+`Config.Keadm.Nodes`: The configuration of the remote nodes to be batch processed.
+
+- `Config.Keadm.Nodes.NodeName`: The name of the remote node. It is not mandatory to be consistent with the actual node name, and it is mainly used to distinguish different node tasks. This parameter must be unique and cannot be repeated.
```

Comment:
The current design does not bind NodeName to `keadm join --edgenode-name`, it is only used as the task name of each node, so that logs can be clearly viewed later